tests/test_suites/ShortSystemTests/test_quota_in_volume_size.sh

```diff
@@ -0,0 +1,238 @@
+sleep 6
+assert_equals ${user_limit} $(get_mountpoint_total_space)
+echo "OK: the total space is exactly ${user_limit_mb}MiB (cache is expired)"
+echo "StatfsCacheTimeout=0" | sudo tee ${info[mount0]}/.saunafs_tweaks
+
+# Check the behavior for group quota
+saunafs setquota -u $(id -u) 0 $((2*user_limit)) 0 0 .
+saunafs setquota -g $(id -g) 0 $((2*user_limit)) 0 0 .
+assert_success sudo -nu saunafstest_0 dd if=/dev/zero of=file2 bs=1M count=$((user_limit_mb))
+sudo -nu saunafstest_0 chown saunafstest_0:$(id -g) file2
+
+# Available space should be 0 because of the group quota: file1 and file2
+# belong to the same group and there is no more space left for that group.
+
+# Check the available space is exactly zero
+assert_equals $(get_mountpoint_available_space) 0
+echo "Ok: the available space is exactly zero"
+
+# Delete the files to free up space
+rm -f file1 file2
+
+sleep 2
+
+# Set user and group quotas to user_limit
+saunafs setquota -u $(id -u) 0 $((user_limit)) 0 0 .
+saunafs setquota -g $(id -g) 0 $((user_limit)) 0 0 .
+
+# Check the behavior for folder quota
+# Test increasing directory quota beyond user quota
+# Create a folder with a quota of 1/2 user_limit
+mkdir folder2
+saunafs setquota -d 0 $((user_limit / 2)) 0 0 folder2
+
+# Verify folder quota is enforced
+assert_success dd if=/dev/zero of=folder2/file4 bs=1M count=$((user_limit_mb / 2))
+assert_failure dd if=/dev/zero of=folder2/file4 bs=1M count=$((user_limit_mb))
+
+# Increase folder quota to exceed user quota
+saunafs setquota -d 0 $((2 * user_limit)) 0 0 folder2
+
+# Verify directory quota is enforced
+assert_failure dd if=/dev/zero of=folder2/file4 bs=1M count=$((user_limit_mb + 1))
+
+# Check the available space is exactly zero
+assert_equals $(get_mountpoint_available_space) 0
+echo "Ok: the available space is exactly zero"
+
+# Delete the files to free up space
+rm -rf folder2
+
+sleep 2
+
+# Set user and group quotas to user_limit
+saunafs setquota -u $(id -u) 0 $((user_limit)) 0 0 .
+saunafs setquota -g $(id -g) 0 $((user_limit)) 0 0 .
+
+# Test directory quota on nested directories
+# Create nested folders with different quotas
+mkdir -p folder3/subfolder1
+saunafs setquota -d 0 $((user_limit * 3 / 4)) 0 0 folder3
+saunafs setquota -d 0 $((user_limit / 2)) 0 0 folder3/subfolder1
+
+# Verify quotas are enforced at each level
+assert_success dd if=/dev/zero of=folder3/file5 bs=1M count=$((user_limit_mb / 2))
+assert_success dd if=/dev/zero of=folder3/subfolder1/file6 bs=1M count=$((user_limit_mb / 4))
+assert_failure dd if=/dev/zero of=folder3/subfolder1/file6 bs=1M count=$((user_limit_mb / 2))
+
+# Last operation should have failed after writing a full chunk
+# Available space should (user_limit_mb - user_limit_mb / 2 - 64)MB
+assert_less_than $(get_mountpoint_available_space) \
+    $((user_limit - user_limit / 2 - SAUNAFS_CHUNK_SIZE + 1))
+echo "Ok: the available space is as expected"
+
+# Delete the files to free up space
+rm -rf folder3/subfolder1 folder3
```

Comment:
never use `-f` when not needed
```suggestion
rm -r folder3/subfolder1 folder3
```

Reply:
Done.